### PR TITLE
Deprecated component props in favor of type helper

### PR DESCRIPTION
**Pull request by MBudreviciusBentley**
## Changes

Deprecated `@public` and `@alpha` component props in favor of type helper, when props are only used for a single component,

## Testing

N/A

### CODE REVIEW

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -3,10 +3,50 @@
...
 
 - [@itwin/appui-react](#itwinappui-react)
+  - [Deprecations](#deprecations)
   - [Changes](#changes)
 
 ## @itwin/appui-react
 
+### Deprecations
+
+- Deprecated `SignInProps` instead use `React.ComponentProps<typeof SignIn>`. [#991](https://github.com/iTwin/appui/pull/991)
```
Comment (GerardasB) [inline docs/changehistory/NextVersion.md:13]: Let's join these under a single bullet point (per package) as it is repetitive and bloats the changelog.

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -3,10 +3,50 @@
...
+- Deprecated `AccuDrawDialogProps` instead use `React.ComponentProps<typeof AccuDrawDialog>`. [#991](https://github.com/iTwin/appui/pull/991)
```
Comment (GerardasB) [inline docs/changehistory/NextVersion.md:14]: Let's add one code snippet example as well.

File: docs/changehistory/NextVersion.md [hunk 3]
```diff
@@ -3,10 +3,50 @@
...
+- Deprecated `AccuDrawFieldContainerProps` instead use `React.ComponentProps<typeof AccuDrawFieldContainer>`. [#991](https://github.com/iTwin/appui/pull/991)
+- Deprecated `AccuDrawInputFieldProps` instead use `React.ComponentProps<typeof AccuDrawInputField>`. [#991](https://github.com/iTwin/appui/pull/991)
```
Comment (GerardasB) [inline docs/changehistory/NextVersion.md:16]: Not all of these types are from the `@itwin/appui-react` package.

**Review by GerardasB — CHANGES_REQUESTED** [review]

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md:13]: Should I still mention what props where deprecated? Or just summarize that various props were deprecated in favor of `ComponentProps`?

File: docs/changehistory/NextVersion.md (hunk 2 above)
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md:14]: Is one snippet enough to summarize all changes. Or a code snippet for each type of props?

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md]: Grouped them by packages.

File: docs/changehistory/NextVersion.md (hunk 2 above)
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md]: Added a code snippet to every package group.

File: docs/changehistory/NextVersion.md (hunk 3 above)
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md]: Split props by the packages.

File: docs/changehistory/NextVersion.md [hunk 4]
```diff
@@ -3,13 +3,33 @@
...
+
+  ```tsx
+  import { AccuDrawDialog } from "@itwin/appui-react";
+
+  type AccuDrawDialogProps = React.ComponentProps<typeof AccuDrawDialog>;
+
+  function UsingAccuDrawDialog(props: AccuDrawDialogProps) {
+    const [opened, setOpened] = React.useState(props.opened);
+    // ...
+  }
```
[the remark this answers is not in the document]
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md]: Done.

File: docs/changehistory/NextVersion.md [hunk 5]
```diff
@@ -23,3 +43,36 @@ Table of contents:
...
+
+  ```tsx
+  import { FormatUnits } from "@itwin/imodel-components-react";
+
+  type FormatUnitsProps = React.ComponentProps<typeof FormatUnits>;
+
+  function UseFormatUnits(props: FormatUnitsProps) {
+    const { initialFormat, persistenceUnit, unitsProvider, onUnitsChange } = props;
+    // ...
+  }
```
[the remark this answers is not in the document]
Comment (MBudreviciusBentley) [reply docs/changehistory/NextVersion.md]: Done.

**Pull request merged**